### PR TITLE
feat(reindex): Implement Reindex status Endpoint

### PR DESCRIPTION
**Pull request by viacheslavkol**
### Purpose
Implement Reindex status Endpoint

### Approach
- Implement DB schema
- Implement status update trigger on processed items updates
- Implement endpoint

### Changes Checklist
- [x] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [x] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [x] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [x] **Permissions**: Document any changes to permissions.
- [ ] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [ ] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [ ] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
[MSEARCH-797](https://folio-org.atlassian.net/browse/MSEARCH-797)

### CODE REVIEW

**Review by viacheslavpoliakov — APPROVED** [review]

File: src/main/resources/changelog/changes/v4.0/create-reindex-entity-tables.xml [hunk 1]
```diff
@@ -229,6 +229,33 @@
...
+      </column>
+      <column name="total_merge_ranges" type="int eger"/>
+      <column name="processed_merge_ranges" type="integer"/>
+      <column name="total_upload_ranges" type="integer"/>
+      <column name="processed_upload_ranges" type="integer"/>
+      <column name="start_time_merge" type="DATETIME"/>
+      <column name="end_time_merge" type="DATETIME"/>
+      <column name="start_time_upload" type="DATETIME"/>
+      <column name="end_time_upload" type="DATETIME"/>
+    </createTable>
```
Comment (mukhiddin-yusuf) [inline src/main/resources/changelog/changes/v4.0/create-reindex-entity-tables.xml:256]: how about tenant id column?

File: src/main/resources/changelog/changes/v4.0/create-reindex-entity-tables.xml [hunk 2]
```diff
@@ -229,6 +229,33 @@
...
   </changeSet>
 
+  <changeSet id="MSEARCH-797@@create-reindex_status-table" author="Viacheslav_Kolesnyk">
+    <preConditions>
+      <not>
+        <tableExists tableName="reindex_status"/>
+      </not>
+    </preConditions>
+
+    <comment>Create upload_ranges table</comment>
```
Comment (mukhiddin-yusuf) [inline src/main/resources/changelog/changes/v4.0/create-reindex-entity-tables.xml:239]:
```suggestion
    <comment>Create reindex_status table</comment>
```

File: src/main/java/org/folio/search/model/reindex/ReindexStatusEntity.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  public static final String PROCESSED_MERGE_RANGES_COLUMN = "processed_merge_ranges";
+  public static final String TOTAL_UPLOAD_RANGES_COLUMN = "total_upload_ranges";
+  public static final String PROCESSED_UPLOAD_RANGES_COLUMN = "processed_upload_ranges";
+  public static final String START_TIME_MERGE_COLUMN = "start_time_merge";
+  public static final String END_TIME_MERGE_COLUMN = "end_time_merge";
+  public static final String START_TIME_UPLOAD_COLUMN = "start_time_upload";
+  public static final String END_TIME_UPLOAD_COLUMN = "end_time_upload";
+
+  private final ReindexEntityType entityType;
+  private ReindexStatus status;
```
Comment (mukhiddin-yusuf) [inline src/main/java/org/folio/search/model/reindex/ReindexStatusEntity.java:24]: ```status``` column is not nullable so maybe makes sense to require to set status field at instance creation

File: src/main/resources/swagger.api/examples/result/ReindexStatusResult.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    endTimeUpload: '2024-04-01T01:37:37.15755006Z'
+  - entityType: 'subject'
+    status: 'Merge In Progress'
+    totalMergeRanges: 3
+    processedMergeRanges: 2
+    startTimeMerge: '2024-04-01T01:37:34.15755006Z'
+  - entityType: 'contributor'
+    status: 'Merge Completed'
+    totalMergeRanges: 3
+    processedMergeRanges: 3
```
Comment (mukhiddin-yusuf) [inline src/main/resources/swagger.api/examples/result/ReindexStatusResult.yaml:20]: we do not set merge status and counts for ```contributor``` and same for ```subject```, ```classification```

File: src/main/resources/changelog/changes/v4.0/create-reindex-entity-tables.xml (hunk 1 above)
Comment (psmagin) [reply src/main/resources/changelog/changes/v4.0/create-reindex-entity-tables.xml:256]: We don't need tenant id for status.

**Review by mukhiddin-yusuf — APPROVED** [review]

**Review by psmagin — APPROVED** [review]

**Pull request merged**